app/miragevpn_client_lwt.ml

```diff
@@ -15,29 +19,43 @@ let platform =
 
 let connected_ip = ref Ipaddr.V4.any
 
-let remote_host_route dst =
+let shares_subnet dst =
   let dst = Ipaddr.V4.to_string dst in
   match Lazy.force platform with
   | Linux -> (
       let cmd = Bos.Cmd.(v "ip" % "route" % "get" % dst) in
       match Bos.OS.Cmd.(run_out cmd |> out_string |> success) with
       | Ok routes -> (
           match String.split_on_char ' ' routes with
-          | _ :: "via" :: ip :: _ -> Some ip
-          | _ -> None)
+          | _ :: "via" :: _ip :: _ -> false
+          | _ -> true)
       | Error (`Msg m) -> invalid_arg ("couldn't find default route " ^ m))
   | FreeBSD -> (
       let cmd = Bos.Cmd.(v "route" % "-n" % "show" % dst) in
       match Bos.OS.Cmd.(run_out cmd |> out_lines |> success) with
-      | Ok lines -> (
-          match
-            List.find_opt
-              (fun l -> String.starts_with ~prefix:"gateway:" (String.trim l))
-              lines
-          with
-          | Some gw ->
-              Some (String.trim (List.nth (String.split_on_char ':' gw) 1))
-          | None -> None)
+      | Ok lines ->
+          not
+            (List.exists
+               (fun l -> String.starts_with ~prefix:"gateway:" (String.trim l))
+               lines)
+      | Error (`Msg m) -> invalid_arg ("couldn't find default route " ^ m))
+
+let default_route () =
+  match Lazy.force platform with
+  | Linux -> (
+      let cmd = Bos.Cmd.(v "ip" % "route" % "show" % "default") in
+      match Bos.OS.Cmd.(run_out cmd |> out_string |> success) with
+      | Ok ip -> (
+          match String.split_on_char ' ' ip with
+          | "default" :: "via" :: ip :: _ -> Some ip
+          | _ -> None)
+      | Error (`Msg m) -> invalid_arg ("couldn't find default route " ^ m))
+  | FreeBSD -> (
+      let cmd = Bos.Cmd.(v "route" % "-n" % "show" % "default") in
+      match Bos.OS.Cmd.(run_out cmd |> out_lines |> success) with
+      | Ok (_ :: _ :: _ :: gw :: _) ->
+          Option.map String.trim (List.nth_opt (String.split_on_char ':' gw) 1)
+      | Ok _ -> None
```

Comment:
```suggestion
    | Ok lines -> (
          match
            List.find_opt
              (fun l -> String.starts_with ~prefix:"gateway:" (String.trim l))
              lines
          with
          | Some gw ->
              Some (String.trim (List.nth (String.split_on_char ':' gw) 1))
          | None -> None)
```

I find the String.nth and `_::_::_::gw::_` a bit too handwavy (and a bit too tied to the current output of route).